## parameter for px4_multidrone_pos_controller.cpp
## 起飞高度
Takeoff_height : 0.3
Disarm_height : 0.3
## 1 for use the accel command
Use_accel : 0.0
## 1 for printf the state, 0 for block
PrintState : true 

##ppn_kx : 0.2
##ppn_ky : 0.2
##ppn_kz : 0.2

##noise_a_xy : 0.0
##noise_b_xy : 0.00

##noise_a_z : 0.0
##noise_b_z : 0.00
##noise_T : 1.0

##noise_start_time : 100.0
##noise_end_time : 30.0

## parameter for px4_pos_estimator.cpp
pos_estimator:
## 使用激光SLAM数据orVicon数据 0 for vicon， 1 for 激光SLAM
  flag_use_laser_or_vicon : 0
## 0 for use the data from fcu, 1 for use the mocap raw data(only position), 2 for use the mocap raw data(position and velocity)
  Use_mocap_raw : 0
  linear_window : 3
  angular_window : 3
  noise_a : 0.0
  noise_b : 0.0
  noise_T : 1.0

## 飞机参数
Quad:
  mass: 1.6
## payload mass
Payload:
  mass: 0.5
## cable length
Cable:
  length: 0.91

## single or multi-UAV mode
CooperativeMode:
  isMulti: false
  droneID: 2              # single UAV payload mode is activated in drone 2

ActionMode:
  type: 1                  # 0 for circle path, 1 for rectangular path

##geo fence for fsc
geo_fence:
  x_min: -1.6
  x_max:  1.5
  y_min: -1.3
  y_max:  1.3
  z_min: -0.05
  z_max:  2.5

## 限制参数
Limit:
  pxy_error_max: 10.0
  vxy_error_max: 10.0
  pz_error_max: 10.0
  vz_error_max: 10.0
  pxy_int_max : 10.0
  pz_int_max : 10.0
  tilt_max: 20.0
  int_start_error: 10.0

  XY_VEL_MAX : 10.0
  Z_VEL_MAX : 10.0

# 位置环参数 - cascade pid
Pos_cascade_pid:
   Kp_xy : 0.95
   Kp_z : 1.0
   Kp_vxvy : 0.09
   Kp_vz : 0.2
   Ki_vxvy : 0.02
   Ki_vz : 0.02
   Kd_vxvy : 0.01
   Kd_vz : 0.0
   Hover_throttle : 0.45
   MPC_VELD_LP: 5.0

# 位置环参数 - normal pid
Pos_pid:
  Kp_xy : 2.5
  Kp_z : 2.5
  Ki_xy : 0.5
  Ki_z : 0.5
  Kd_xy : 3.0
  Kd_z : 3.0


# 位置环参数 for ude
Pos_ude:
  Kp_xy : 0.5
  Kp_z : 0.5
  Kd_xy : 2.0
  Kd_z : 2.0
  T_ude_xy : 2.0
  T_ude_z : 2.0

# 位置环参数 for passivity
Pos_passivity:
  Kp_xy : 0.5
  Kp_z : 0.5
  Kd_xy : 2.0
  Kd_z : 2.0
  T_ude_xy : 1.0
  T_ude_z : 1.0
  T_ps: 0.2

# 位置环参数 for ne
Pos_ne:
  Kp_xy : 0.5
  Kp_z : 0.5
  Kd_xy : 2.0
  Kd_z : 2.0
  T_ude_xy : 1.0
  T_ude_z : 1.0
  T_ne : 0.1

# 圆轨迹参数 for Circle_Trajectory
Circle_Trajectory:
  Center_x: 0.0
  Center_y: 0.0
  Center_z: 1.5
  radius: 1.0
  linear_vel: 0.1
  time_total: 50.0
  direction: 1.0

# parameter for rectangular trajectory
Rectangular_Trajectory:
  a_x: 2.0 # length along x axis
  a_y: 2.0 # length along y axis
  h: 1.5 # hight
  vel_x: 0.1 # velocity along x axis
  vel_y: 0.1 # velocity along y axis 
  center_x: 0.0
  center_y: 0.0
  center_z: 0.0

payload_geofence:
  x_min: -0.6
  x_max:  0.6
  y_min: -0.4
  y_max:  0.4
  z_min: -0.05
  z_max:  0.6

# cooperative method
CooperativePayload: 1
IntegrationStartHeight: 0.15

# parameter for single UAV payload using method in TIE paper
Pos_tie:
  issingleUAVused: true   # is single UAV payload used in control
  isVisionused: false
  isIntegrationOn: true
  isPubAuxiliaryState: true
  payload_mass: 0.5
  quadrotor_mass : 1.63
  cablelength : 1
  motor_slope: 0.082
  motor_intercept: 0.26
  Kp_xy : 0.5
  Kp_z :  0.7
  Kv_xy : 2.0
  Kv_z :  2.0
  T_tie_xy : 0.8
  T_tie_z : 0.6
  Kpv_xy: 0.0
  Kpv_z: 0.0
  KL : 0.0

# parameter for payload cooperative control
uav0_Pos_GNC:
  mass : 1.55
  cablelength : 0.91
  TetherOffset_x: 0.99
  TetherOffset_y: 0
  TetherOffset_z: 0
  PayloadSharingPortion : 0.5
  motor_slope: 0.082
  motor_intercept: 0.26


uav1_Pos_GNC:
  mass : 1.55
  cablelength : 0.91
  TetherOffset_x: -0.99
  TetherOffset_y: 0
  TetherOffset_z: 0
  PayloadSharingPortion: 0.5
  motor_slope: 0.082
  motor_intercept: 0.26

uav2_Pos_GNC:
  mass : 1.55
  cablelength : 0
  TetherOffset_x: 0
  TetherOffset_y: 0
  TetherOffset_z: 0
  PayloadSharingPortion: 0
  motor_slope: 0.082
  motor_intercept: 0.26

uav3_Pos_GNC:
  mass : 1.45
  cablelength : 0
  TetherOffset_x: 0
  TetherOffset_y: 0
  TetherOffset_z: 0
  PayloadSharingPortion: 0
  motor_slope: 0.082
  motor_intercept: 0.26

Pos_GNC:
  ControllerInfo: -1
  Type: 0              # 0 for TCST method
  MaxInclination: 50.0 # maximum safty angle
  CableLengthTolerance: 1.2
  # debugging options
  UseAddonForce: true
  UseCrossFeedingTerms: true
  PubAuxiliaryState : true
  # controller parametrers:
  num_drone : 2
  kv_xy : 0.25
  Kv_z  : 0.7
  kR_xy : 0.3
  kR_z  : 0.2
  kL    : 0.15
  Kphi_xy : 3
  Kphi_z  : 4.5
  fp_max_x : 5
  fp_max_y : 5
  fp_max_z : 5
  lambda_j : 0.30
  lambda_Txy : 0.08
  lambda_Tz  : 0.3
  lambda_Rxy : 0.09
  lambda_Rz  : 0.09
  kF_xy : 0.4
  kF_z :  0.5
  kr1_x : 0.15
  kr1_y : 0.15
  kr1_z : 0.15
  kr2_x : 0.15
  kr2_y : 0.15
  kr2_z : 0.15

  kp_x : 0.1
  kp_y : 0.1
  kp_z : 0.1

  komega_x : 0.1
  komega_y : 0.1
  komega_z : 0.1

  lambda1_x : 0.3
  lambda1_y : 0.3
  lambda1_z : 0.3
  lambda2_x : 0.3
  lambda2_y : 0.3
  lambda2_z : 0.3
  pos_int_max_x: 10
  pos_int_max_y: 10
  pos_int_max_z: 10
  ang_int_max_x : 1
  ang_int_max_y : 1
  ang_int_max_z : 1
  pos_int_start_error: 0.3
  angle_int_start_error: 0.5
  ## temporary disabled: (integral gain for payload stabilization)
  kvi_xy : 0.00
  kvi_z  : 0.00
  kRi_xy : 0.0
  kRi_z  : 0.0

Pos_JGCD:
  MaxInclination: 50.0 # maximum safty angle
  CableLengthTolerance: 1.2
  PubAuxiliaryState : true
  # controller parametrers:
  num_drone : 3
  kv_xy : 0.3
  Kv_z  : 0.7
  kR_xy : 0.3
  kR_z  : 0.2
  kL    : 0.12
  Kphi_xy : 3.0
  Kphi_z  : 4.5
  fp_max_x : 10.0
  fp_max_y : 10.0
  fp_max_z : 10.0
  lambda_j : 1.0